### PR TITLE
[CatalogPromotions] Use actions for discount application

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | based on https://github.com/Sylius/Sylius/pull/13029
| License         | MIT

### CODE REVIEW

**Review by arti0090 — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Processor/DummyCatalogPromotionProcessor.php [hunk 1]
```diff
@@ -16,9 +16,9 @@
 use Doctrine\ORM\EntityManagerInterface;
 use Sylius\Bundle\CoreBundle\Applicator\CatalogPromotionApplicatorInterface;
 use Sylius\Component\Core\Model\CatalogPromotionInterface;
-use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Sylius\Component\Core\Provider\CatalogPromotionVariantsProviderInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
 
 final class DummyCatalogPromotionProcessor implements CatalogPromotionProcessorInterface
```
Comment (AdamKasp) [inline src/Sylius/Bundle/CoreBundle/Processor/DummyCatalogPromotionProcessor.php:23]: we have merged rules, and now we are merging action, is it still Dummy processor?

Comment (Zales0123) [reply src/Sylius/Bundle/CoreBundle/Processor/DummyCatalogPromotionProcessor.php:23]: It's not dummy anymore! 🎉 🎉 🎉

File: src/Sylius/Behat/Context/Api/Admin/ManagingCatalogPromotionsContext.php [hunk 2]
```diff
@@ -222,6 +255,27 @@ public function thereShouldBeNewCatalogPromotionOnTheList(int $amount = 0): void
...
+     */
+    public function itShouldHaveDiscount(float $amount): void
+    {
+        $catalogPromotion = $this->responseChecker->getCollection($this->client->getLastResponse())[0];
+
+        Assert::same($catalogPromotion['actions'][0]['configuration']['amount'], $amount);
+    }
+
+    /**
+     * @Then /^this catalog promotion should have ("[^"]+") percentage discount$/
```
[the remark this answers is not in the document]
Comment (Zales0123) [reply src/Sylius/Behat/Context/Api/Admin/ManagingCatalogPromotionsContext.php:269]: Good idea, I've changed that 👍

File: src/Sylius/Behat/Context/Api/Admin/ManagingCatalogPromotionsContext.php [hunk 3]
```diff
@@ -165,7 +165,7 @@ public function iWantToModifyACatalogPromotion(CatalogPromotionInterface $catalo
     /**
      * @When /^it gives the ("[^"]+") percentage discount$/
      */
-    public function iAddThePercentageDiscountCatalogPromotionActionConfiguredWithAmountOf(float $amount): void
+    public function iGivesPercentageDistount(float $amount): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingCatalogPromotionsContext.php:168]:
```suggestion
    public function itGivesPercentageDiscount(float $amount): void
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Mateusz! :1st_place_medal:

**Pull request merged**